src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,390 @@
+'use strict';
+
+const Base = require('./Base');
+const {
+  GuildScheduledEventEntityTypes,
+  GuildScheduledEventStatuses,
+  GuildScheduledEventPrivacyLevels,
+  Endpoints,
+} = require('../util/Constants');
+const SnowflakeUtil = require('../util/SnowflakeUtil');
+
+/**
+ * Represents a scheduled event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildScheduledEvent extends Base {
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The id of the guild scheduled event
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The id of the guild this guild scheduled event belongs to
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
+
+    /**
+     * Whether the guild scheduled event has been deleted
+     * @type {boolean}
+     */
+    this.deleted = false;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    if ('channel_id' in data) {
+      /**
+       * The channel id in which the scheduled event will be hosted, or `null` if entity type is `EXTERNAL`
+       * @type {?Snowflake}
+       */
+      this.channelId = data.channel_id;
+    } else {
+      this.channelId ??= null;
+    }
+
+    if ('creator_id' in data) {
+      /**
+       * The id of the user that created this guild scheduled event
+       * @type {?Snowflake}
+       */
+      this.creatorId = data.creator_id;
+    } else {
+      this.creatorId ??= null;
+    }
+
+    /**
+     * The name of the guild scheduled event
+     * @type {string}
+     */
+    this.name = data.name;
+
+    if ('description' in data) {
+      /**
+       * The description of the guild scheduled event
+       * @type {?string}
+       */
+      this.description = data.description;
+    } else {
+      this.description ??= null;
+    }
+
+    /**
+     * The timestamp the guild scheduled event will start at
+     * @type {number}
+     */
+    this.scheduledStartTime = new Date(data.scheduled_start_time).getTime();
```

Comment:
This is not sent when fetching a removed guild scheduled event from the audit logs and will return `NaN`, so I would suggest making this nullable just like `scheduledEndTime`.

I noticed this is documented as not nullable in Discord's API documentation, so it's probably not documented correctly there too?